tests/conftest.py

```diff
@@ -1,6 +1,72 @@
 from .common import *
 
 
+@pytest.fixture(scope="session")
+def adata_pbmc():
+    #adata_ref = sc.datasets.pbmc3k_processed()
+    # quick fix for broken dataset paths, should be removed with scanpy>=1.6.0
+    adata_ref = sc.read(
+        "pbmc3k_processed.h5ad",
+        backup_url="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/main/example-dataset/pbmc3k.h5ad"
+    )
+    adata = sc.datasets.pbmc68k_reduced()
+
+    var_names = adata_ref.var_names.intersection(adata.var_names)
+    adata_ref = adata_ref[:, var_names]
+    adata = adata[:, var_names]
+
+    sc.pp.pca(adata_ref)
+    sc.pp.neighbors(adata_ref)
+    sc.tl.umap(adata_ref)
+
+    # merge cell type labels
+    sc.tl.ingest(adata, adata_ref, obs='louvain')
+    adata_concat = adata_ref.concatenate(adata, batch_categories=['ref', 'new'])
+    adata_concat.obs.louvain = adata_concat.obs.louvain.astype('category')
+    # fix category ordering
+    adata_concat.obs.louvain.cat.reorder_categories(adata_ref.obs.louvain.cat.categories, inplace=True)
+    adata_concat.obs['celltype'] = adata_concat.obs['louvain']
+
+    del adata_concat.obs['louvain']
+    del adata_concat.uns
+    del adata_concat.obsm
+    del adata_concat.varm
+
+    yield adata_concat
+    del adata_concat
+
+
+@pytest.fixture(scope="function")
+def adata(adata_pbmc):
+    adata = adata_pbmc.copy()
+    yield adata
+    del adata
```

Comment:
`scope=function` is redundant here, right? But good to leave I guess as it is more explicit. The scope ensures that a different copy is received by every test function if I understand correctly?

Reply:
Naming here is not really great... as this fixture can be confused with the object name. Maybe rename to `adata_pbmc_factory` or so.